### PR TITLE
Fix: prevent sign waiver from showing when user is confirmed and properly update vertical timeline

### LINKED ISSUE #253: [Portal] Sign Waiver shown when confirmed and Vertical Timeline not updated
When user's status is `CONFIRMED`, the Vertical Timeline still shows an empty circle and says "Confirm attendance", and the `SignWaiver` component is shown when it shouldn't be.

The conditional status was not implemented properly in #244.

### PR DESCRIPTION
**Pull request by waalbert**
Resolves #253

### CODE REVIEW

File: apps/site/src/app/(main)/portal/@applicant/SignWaiver.tsx [hunk 1]
```diff
@@ -7,7 +7,8 @@ interface SignWaiverProps {
 }
 
 function SignWaiver({ status }: SignWaiverProps) {
-	return status === PortalStatus.waived ? null : (
+	return status === PortalStatus.waived ||
+		status === PortalStatus.confirmed ? null : (
```
Comment (taesungh) [inline apps/site/src/app/(main)/portal/@applicant/SignWaiver.tsx:11]: Chore: we already conditionally render this `SignWaiver` component from `ApplicantPortal`, so could we fix the condition there and remove the condition here?

File: apps/site/src/app/(main)/portal/@applicant/ApplicantPortal.tsx [hunk 2]
```diff
@@ -28,14 +28,14 @@ async function Portal() {
...
-	const isAccepted = status === PortalStatus.accepted || submittedWaiver;
+	const isAccepted = status === PortalStatus.accepted;
 
 	return (
 		<div className="bg-white text-black max-w-4xl rounded-2xl p-6 flex flex-col mb-24 w-full">
 			<h2 className="text-4xl font-semibold">Status</h2>
 			<VerticalTimeline status={status as PortalStatus} />
 			<Message status={status as PortalStatus} />
-			{isAccepted && <SignWaiver status={status as PortalStatus} />}
+			{isAccepted && !submittedWaiver && <SignWaiver />}
```
Comment (taesungh) [inline apps/site/src/app/(main)/portal/@applicant/ApplicantPortal.tsx:38]: Thought: all of the states are mutually exclusive, so we shouldn't need the second conditional. Perhaps consider renaming `isAccepted` to `needsToSignWaiver`?

**Review by samderanova — APPROVED** [review]
Thanks for simplifying the conditionals and fixing the issue.

**Review by taesungh — APPROVED** [review]
Thanks for fixing this.

**Pull request merged**